external/textprogressbar/textprogressbar.m

```diff
@@ -17,16 +17,18 @@ function textprogressbar(c)
 %
 %      - renamed strCR to WAITBAR_HANDLE
 %      - changed persistent type of strCR to global type
+%      - Changes To the progressbar text to be consistent (to allow
+%        intermittent output)
 
 %% Initialization
 global WAITBAR_HANDLE;           %   Carriage return pesistent variable
-
+persistent WAITBAR_PROGRESS;
```

Comment:
Persistent is entirely enough. This is a field, that is only ever called in this function, and therefore global would be unnecessary. From my understanding, persistent variables can be considered as private fields of a function handle (i.e. the function is treated as an object and the persistent variable is a field of the function object). The assumption in this context is, that there is only ever one textprogressbar at the same time (we might actually consider to change this to an object at some point), and as such that object uses the given String.